frontend/public/components/utils/resource-log.jsx

```diff
@@ -0,0 +1,249 @@
+const LogControls = (props) => {
+  return <div className="co-m-pane__top-controls">
+    { props.logState === 'loading' && <span className="co-icon-space-l"><LoadingInline /></span> }
+    { ['streaming', 'paused', 'eof'].includes(props.logState) && <TogglePlay active={props.logState === 'streaming'} onClick={props.toggleStreaming} disabled={props.logState === 'eof'}/> }
+    <span className="log-container-selector__text">
+      { props.logState === 'loading' && <span>Loading log...</span> }
+      { props.logState === 'streaming' && <span>Streaming logs from</span> }
+      { props.logState === 'paused' && <span>Log stream is paused.</span> }
+      { props.logState === 'eof' && <span>Log stream ended from </span> }
+    </span>
+    <ContainerSelect {...props} />
+  </div>;
+};
+
+export class ResourceLog extends SafetyFirst {
+  constructor(props) {
+    super(props);
+
+    this._buffer = lineBuffer(1000);
+    this._loadTime = Date.now();
+    this._pendingReload = null;
+
+    this._touchLoadTimeState = _.throttle(this._touchLoadTimeState, 100);
+    this._updateLogState = this._updateLogState.bind(this);
+    this._toggleStreaming = this._toggleStreaming.bind(this);
+
+    this.state = {
+      logURL: '',
+      loadTime: 0,
+      logState: 'loading'
+    };
+
+    this.state = _.defaults({}, this._initialState(), this.state);
+  }
+
+  componentDidMount() {
+    super.componentDidMount();
+    this._beginStreaming();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // If container changed, restart streaming
+    if (nextProps.container !== this.props.container) {
+      this._endStreaming();
+      this._buffer = lineBuffer(100);
+      this.setState(this._initialState(nextProps), this._beginStreaming);
+    } else {
+      this.setState(this._initialState(nextProps));
+    }
+  }
+
+  componentWillUnmount() {
+    super.componentWillUnmount();
+    this._endStreaming();
+  }
+
+  _initialState(props = this.props) {
+    const newState = {};
+    const namespace = _.get(props.obj, 'metadata.namespace');
+    const objName = _.get(props.obj, 'metadata.name');
+
+    // If the container changed, reset log state to loading, otherwise keep it as is
+    const initialLogState = this.props.container !== props.container ? 'loading' : this.state.logState;
+
+    // If log is finished, change to eof log state
+    newState.logState = props.eof ? 'eof' : initialLogState;
+
+    newState.logURL = this._logURL(props.model, namespace, objName, props.currentResource);
+    return newState;
+  }
+
+  // Returns API url for given resource log
+  _logURL(model, namespace, name, container = '') {
+    return resourceURL(model, {
+      ns: namespace,
+      name,
+      path: 'log',
+      queryParams: {
+        container: container,
+        follow: 'true',
+        tailLines: this._buffer.maxSize
+      }
+    });
+  }
+
+  // Updates log state as long as eof has not been reached
+  _updateLogState(newState) {
```

Comment:
Only update `logState` if `eof` property is false. Prevents changing the logState when it is no longer streaming.